### ISSUE TITLE
[.NET11][WinUI] Accessibility/Keyboard Navigation doesnot works on CollectionView2

### ISSUE THREAD

**Issue by SuthiYuvaraj** [dotnet/maui#38660]
### Description

In CollectionView2 on WinUI does not provide the expected accessibility and keyboard navigation experience. Unable to reliably focus and navigate items using Tab navigation, and screen readers do not announce Semantic Description and Semantic Hint information for individual items.

### Steps to Reproduce

1.Run the attached sample
2.Press Tab and move to the CollectionView 

Expected Behaviour:
CollectionView should focus on tab and navigate between the items with arrow keys 

Actual Behaviour:
Focus skips the collectionview , directly moves to next focusable item 

https://github.com/user-attachments/assets/c1c00be9-22c3-44d0-b955-60aa6562088e

[TestSample.zip](https://github.com/user-attachments/files/32382822/TestSample.zip)

### Version with bug

11.0.0-preview.6

### Is this a regression from previous behavior?

Yes, this used to work in .NET MAUI

### Last version that worked well

11.0.0-preview.5

### Affected platforms

Windows

### Relevant log output

```shell

```

**Comment by Ahamed-Ali** [comment]
This issue has been validated in Visual Studio Professional 2026 Insiders [12120.281] with .NET MAUI versions 11.0.0-Preview 5, .11.0.0-Preview 6 & 11.0.0-Preview 7.. It can be reproduced on windows platform from .Net11 Preview6.
